Mathlib/Analysis/Normed/Group/Basic.lean

```diff
@@ -375,6 +375,14 @@ theorem norm_div_rev (a b : E) : ‖a / b‖ = ‖b / a‖ := by
 @[to_additive (attr := simp) norm_neg]
 theorem norm_inv' (a : E) : ‖a⁻¹‖ = ‖a‖ := by simpa using norm_div_rev 1 a
 
+@[to_additive (attr := simp) norm_abs_zsmul]
+theorem norm_zpow_abs (a : E) (n : ℤ) : ‖a ^ |n|‖ = ‖a ^ n‖ := by
+  rcases le_total 0 n with hn | hn <;> simp [hn, abs_of_nonneg, abs_of_nonpos]
+
+@[to_additive (attr := simp) norm_neg_one_pow_zsmul]
+theorem norm_zpow_neg_one_pow (a : E) (n : ℕ) : ‖a ^ ((-1) ^ n : ℤ)‖ = ‖a‖ := by
```

Comment:
Is this one too specific?

Reply:
How about
```lean
theorem norm_zpow_of_is_unit (a : E) (z : ℤ) (hz : IsUnit z) : ‖a ^ z‖ = ‖a‖ := by
```
instead, from which this follows trivially?

Reply:
(this also works nicely if working with `Units ℤ` since you can then use `u.isUnit`)

Reply:
We don't have `Pow _ (Units _)`, so I added an additive-only lemma for `SMul (Units Int) E`.